internal/vm/dir_test.go

```diff
@@ -48,9 +48,9 @@ func TestShimDir(t *testing.T) {
 		{name: "id with ?", ns: "test", id: "?", outErr: `invalid vm id: identifier "?" must match`},
 		{name: "id with *", ns: "test", id: "*", outErr: `invalid vm id: identifier "*" must match`},
 		{name: "id with ,", ns: "test", id: ",", outErr: `invalid vm id: identifier "," must match`},
-		{name: "valid", ns: "ns", id: "1", outDir: "ns/1"},
-		{name: "valid with dashes", ns: "test-123", id: "123-456", outDir: "test-123/123-456"},
-		{name: "valid with dots", ns: "test.123", id: "123.456", outDir: "test.123/123.456"},
+		{name: "valid", ns: "ns", id: "1", outDir: "ns-1/1"},
+		{name: "valid with dashes", ns: "test-123", id: "123-456", outDir: "test-123-123-456/123-456"},
+		{name: "valid with dots", ns: "test.123", id: "123.456", outDir: "test.123-123.456/123.456"},
```

Comment:
It is a bit odd to see the VM ID part is repeated twice. Why do you want to do that?

Reply:
To get a unique namespace dir for each VM.

Reply:
I meant, why don't you use `[namespace][separator][vmID]` instead of `[namespace][separator][vmID]/[vmID]`?

Reply:
Yeah, I can change to `[namespace-vmID]/`. I also have a concern over this design. If we have 100 VMs, then there will be 100 directories for each vm. Does it look reasonable to you? I don't feel confident on this design choice. 
Do you prefer wait all VMs finish and then remove the namespace dir approach?

Reply:
What if the namespace itself contains `nsp`? containerd limits the characters clients can use in namespace names. Please you take a look that first.

Reply:
Another format I am considering is `[namespace.vmID]/`

Reply:
Hmm, why? Seems a namespace itself can contain `.`.

Reply:
I test it on https://regexr.com/
I think it is ok for namespace to contain `.` but not as its suffix. 
You can have `default`, `def.ault`, but not `default.`